src/Pages/Explore.jsx

```diff
@@ -17,41 +19,84 @@ class Explore extends React.Component {
     super(props);
 
     this.state = {
-      loading: true,
+      search: '',
+      loading: false,
       reconstruction: [],
+      total: 0,
+      page: 1,
+      hasPrevious: false,
+      hasNext: false,
     };
-  }
 
-  componentDidMount() {
-    this.fetchReconstructions();
+    this.searchHandler = this.searchHandler.bind(this);
+    this.pageChangeHandler = this.pageChangeHandler.bind(this);
+
+    // debounce
+    this.delayHandler = debounce(this.delayHandler, 1000);
   }
 
   async fetchReconstructions() {
-    let axios = getAxiosInstance();
-    let reconstruction = await axios.get(RECONSTRUCTION_FETCH_URL, {
-      params: {
-        filters: FILTERS,
-        order: SORT_ORDER,
-        size: NUM_RECONSTRUCTIONS,
-      },
-    });
-
     try {
+      this.setState({
+        loading: true,
+      });
+      let axios = getAxiosInstance();
+      let reconstruction = await axios.get(RECONSTRUCTION_FETCH_URL, {
+        params: {
+          q: this.state.search,
+          page: this.state.page,
+          filters: FILTERS,
+          order: SORT_ORDER,
+          size: NUM_RECONSTRUCTIONS,
+        },
+      });
       if (reconstruction.status === STATUS_OK) {
         this.setState({
           loading: false,
           reconstruction: reconstruction.data.data,
+          total: Math.ceil(reconstruction.data.total / NUM_RECONSTRUCTIONS),
+          hasPrevious: reconstruction.data.hasPrevPage,
+          hasNext: reconstruction.data.hasNextPage,
         });
       }
     } catch (err) {
-      if (err) {
-        this.setState({
-          loading: true,
-        });
-      }
+      this.setState({
+        reconstruction: [],
+        loading: false,
+      });
     }
   }
 
+  componentDidMount() {
+    this.fetchReconstructions();
+  }
+
+  delayHandler() {
+    this.fetchReconstructions();
+  }
+
+  searchHandler(e) {
+    this.setState(
+      {
+        search: e.target.value,
```

Comment:
Let's trim the search query.
```suggestion
        search: e.target.value.trim(),
```

Reply:
Ok !